core/src/dird/ua_restore.cc

```diff
@@ -839,15 +845,65 @@ static bool get_date(UaContext* ua, char* date, int date_len)
   ua->SendMsg(
       _("The restored files will the most current backup\n"
         "BEFORE the date you specify below.\n\n"));
+  std::string cmpdate;
   for (;;) {
     if (!GetCmd(ua, _("Enter date as YYYY-MM-DD HH:MM:SS :"))) { return false; }
-    if (StrToUtime(ua->cmd) != 0) { break; }
+    cmpdate = CompensateShortDate(ua->cmd);
+
+    if (StrToUtime(cmpdate.c_str()) != 0) { break; }
     ua->ErrorMsg(_("Improper date format.\n"));
   }
-  bstrncpy(date, ua->cmd, date_len);
+  bstrncpy(date, cmpdate.c_str(), date_len);
```

Comment:
What happens with date after copying to it?

Reply:
It is used within `UserSelectJobidsOrFiles(UaContext* ua, RestoreContext* rx)`.

After calling `get_date(UaContext* ua, char* date, int date_len)`, the result date is used with other functions that expect a date to fetch the actual jobs or data needed before that date, like `SelectBackupsBeforeDate(UaContext* ua, RestoreContext* rx, char* date)` for example.